### ISSUE TITLE
Model to Meaning corrections

### ISSUE THREAD

**Issue by timchisamore** [vincentarelbundock/marginaleffects#1479]
I have identified the following typos/grammatical errors in Section 1 of Model to Meaning:

[Edited to hide details]

<details>

- [ ] 2: Missing “we” - Often, must transform parameter estimates into quantities that directly inform our research questions, and that our audience will readily understand.
- [ ] 3.2: Floating hyperlink - [Section 2.2](https://marginaleffects.com/chapters/challenge.html#sec-goals_estimand) By defining a grid, the analyst gives an explicit answer to this question: what kinds of people or units do estimates apply to?
- [ ] 3.2.5: Broken hyperlinks - This kind of duplication will be essential in chapters [-sec-comparison] and [-sec-gcomputation], where we explore counterfactual analysis and causal inference.
- [ ] 3.4: Counterfactual comparisons instead of comparisons - Chapters [5](https://marginaleffects.com/chapters/predictions.html), [6](https://marginaleffects.com/chapters/comparisons.html), and [7](https://marginaleffects.com/chapters/slopes.html) show how we can apply these strategies to construct standard errors and confidence intervals around predictions, comparisons, and slopes.
- [ ] 3.5: And instead of a - Once we have computed and quantity of interest and its standard error, we can finally conduct a test to see if our hypothesis or conjecture is correct.
- [ ] 3.5: Difference of 0 instead of 1 - Parameter estimates: The difference between the first and second regression coefficients is equal to 1.
- [ ] 3.5: Counterfactual comparisons instead of counterfactual predictions - The marginaleffects package allows us to compute null hypothesis and equivalence tests on raw parameter estimates, (non-)linear combinations of those parameters, as well as on all the quantities estimated by the package: predictions, counterfactual predictions, and slopes
- [ ] 4: Remove “that” - As we will see, these complementary technologies that are extremely versatile; they can be applied to all the quantities of interest studied in this book.
- [ ] 4: Consider changing “drug” to “treatment” to keep it consistent - Is the effect of a new drug different from the effect of an existing treatment?
- [ ] 4: `` instead of “ - We say that a result has ``practical significance” when it has important implications for the real world.
- [ ] 4: Consider changing “we” to “they” - For this article, the authors conducted a randomized controlled trial to find out if we could encourage people to seek information about their HIV status
- [ ] 4: Change “equal is 1” to “equal to 1” - The treatment is a binary variable, incentive, equal is 1 if the participant was part of the treatment group who received an incentive, and 0 if they received no money
- [ ] 4: The agecat variable in the thornton dataset appears to be missing in the table
- [ ] 4.1: “… with a known …” - After choosing , we must pick a test statistic with known sampling distribution, such as t or z.
- [ ] 4.1: Change “compares” to “compare” - We then use observed data to compute the test statistic, and compares it to its assumed distribution under H0.
- [ ] 4.1.2: Change “futher” to “further” - Futher suppose we want to compute ratio of coefficients, instead of differences.
- [ ] 4.2: As above, consider changing “drug” to “treatment” - For example, medical researchers may wish to determine if the effect of a new drug is similar to that of existing treatments, or if it can be considered “negligible” in terms of “clinical significance.”
- [ ] 5.1: Consider clarifying that the values are probabilities - Applying this function ensures that the linear component inside the parentheses of [Equation 5.1](https://marginaleffects.com/chapters/predictions.html#eq-predictions-logit_model) gets rescaled to the interval [0, 1] in order to respect the natural scale of the binary outcome.
- [ ] 5.3: Did you intentionally not mention the 18 to 35 category? - The average predicted probability of seeking one’s test result is about 67% for minors, and 72% for those above 35 years old.
- [ ] 5.3: Did you intentionally not mention the 18 to 35 category? - Now, the average predicted probability of seeking one’s test result is estimated at 54% for minors, and 59% for those above 35 years old
- [ ] 5.3: Consider adding “for” between “values” and “each - To build such a grid, every observation of the dataset is duplicated, and the incentive variable is fixed to different values each counterfactual dataset
- [ ] 5.4: Chapter duplicated - Chapter [Chapter 14](https://marginaleffects.com/chapters/uncertainty.html) devotes substantial space to explain each of these approaches, but it is useful to note that these strategies are implemented using the [inferences()](https://marginaleffects.com/man/r/inferences.html) function
- [ ] 5.6: Consider adding “has” between “which” and “a” - This is easy to do with the [plot_predictions()](https://marginaleffects.com/man/r/plot_predictions.html) function, which a syntax that closely parallels that of the other marginaleffects functions
- [ ] 5.6.1: Consider clarifying that it’s an empirical CDF - This time, however, the y-axis indicates the cumulative distribution, that is, the proportion of data points that are less than or equal to a specific value
- [ ] 5.6.2: Consider removing “actually” - When we use this argument, the function will (1) compute predictions for each observation in the actually observed dataset, and (2) average unit-level predictions across some variable(s) of interest.
- [ ] 5.7: Unsure if you meant to include “attendant” here as it was never mentioned in the section prior - To clearly define predictions and attendant tests, analysts must make five decisions.

</details>

Please let me know if you need any additional information or are unsure of how I have formatted things.

**Comment by timchisamore** [comment]
No problem, as I continue to read I will flag anything else I may notice as well!

**Comment by timchisamore** [comment]
- 4: Remove “that” - As we will see, these complementary technologies that are extremely versatile; they can be applied to all the quantities of interest studied in this book.
- 4: Consider changing “drug” to “treatment” to keep it consistent - Is the effect of a new drug different from the effect of an existing treatment?
- 4: `` instead of “ - We say that a result has ``practical significance” when it has important implications for the real world.
- 4: Consider changing “we” to “they” - For this article, the authors conducted a randomized controlled trial to find out if we could encourage people to seek information about their HIV status
- 4: Change “equal is 1” to “equal to 1” - The treatment is a binary variable, incentive, equal is 1 if the participant was part of the treatment group who received an incentive, and 0 if they received no money
- 4: The agecat variable in the thornton dataset appears to be missing in the table
- 4.1: “… with a known …” - After choosing , we must pick a test statistic with known sampling distribution, such as t or z.
- 4.1: Change “compares” to “compare” - We then use observed data to compute the test statistic, and compares it to its assumed distribution under H0.
- 4.1.2: Change “futher” to “further” - Futher suppose we want to compute ratio of coefficients, instead of differences.
- 4.2: As above, consider changing “drug” to “treatment” - For example, medical researchers may wish to determine if the effect of a new drug is similar to that of existing treatments, or if it can be considered “negligible” in terms of “clinical significance.”
- 5.1: Consider clarifying that the values are probabilities - Applying this function ensures that the linear component inside the parentheses of [Equation 5.1](https://marginaleffects.com/chapters/predictions.html#eq-predictions-logit_model) gets rescaled to the interval [0, 1] in order to respect the natural scale of the binary outcome.
- 5.3: Did you intentionally not mention the 18 to 35 category? - The average predicted probability of seeking one’s test result is about 67% for minors, and 72% for those above 35 years old.
- 5.3: Did you intentionally not mention the 18 to 35 category? - Now, the average predicted probability of seeking one’s test result is estimated at 54% for minors, and 59% for those above 35 years old
- 5.3: Consider adding “for” between “values” and “each - To build such a grid, every observation of the dataset is duplicated, and the incentive variable is fixed to different values each counterfactual dataset
- 5.4: Chapter duplicated - Chapter [Chapter 14](https://marginaleffects.com/chapters/uncertainty.html) devotes substantial space to explain each of these approaches, but it is useful to note that these strategies are implemented using the [inferences()](https://marginaleffects.com/man/r/inferences.html) function
- 5.6: Consider adding “has” between “which” and “a” - This is easy to do with the [plot_predictions()](https://marginaleffects.com/man/r/plot_predictions.html) function, which a syntax that closely parallels that of the other marginaleffects functions
- 5.6.1: Consider clarifying that it’s an empirical CDF - This time, however, the y-axis indicates the cumulative distribution, that is, the proportion of data points that are less than or equal to a specific value
- 5.6.2: Consider removing “actually” - When we use this argument, the function will (1) compute predictions for each observation in the actually observed dataset, and (2) average unit-level predictions across some variable(s) of interest.
- 5.7: Unsure if you meant to include “attendant” here as it was never mentioned in the section prior - To clearly define predictions and attendant tests, analysts must make five decisions.

**Comment by vincentarelbundock** [comment]
These are great, thanks!

**Comment by timchisamore** [comment]
- 8.2: There should be a value instead of NA - In contrast, for individuals who were employed (work=1), the estimated effect is NA, meaning that winning a big lottery prize significantly reduces labor earnings.
- 9.1: Remove one of “be” or “remain” - Hence, the workflow described below would be remain unchanged if the analyst chose to fit a GLM (or other) model instead of this LPM.
- 10.1.1: Switch “explanator” to “explanatory variable” or some synonym - The first case to consider is when the association between a categorical explanator and an outcome is moderated by a categorical variable .
- 10.1.1: Switch “squares” to “circles” - The triangles in this figure are systematically above the corresponding squares.
- 10.1.1: Switch “squares” to “circles” - For some values of M, the triangles are farther apart from the squares, suggesting that the effect of X on M may be stronger when takes on those values.
- 10.1.2: Change “explanator” to “explanatory variable” or some synonym - The second case to consider is an interaction between a categorical explanator (X) and a continuous mediator (M). To illustrate, we consider a new simulated dataset.
- 10.1.2: Add “on” between “effect” and “the” - We can reject the null hypothesis that M has no moderating effect the relationship between X and Y.
- 10.1.3: Remove “plot” - [Figure 10.4](https://marginaleffects.com/chapters/interactions.html#fig-interactions_plot_slopes_num_num) plot shows that when the moderator M is below -1, the relationship between X and Y is negative: increasing decreases .
- 10.1.4: Remove one of “check” or “estimate” - To check estimate if the effect of X varies depending on the value of moderator M1, we call [avg_comparisons()](https://marginaleffects.com/man/r/comparisons.html) with the by argument.
- 10.2: Consider changing “… when the n order …” with “when the degree …” - One significant issue is its potential for overfitting, especially when the n order is high.
- 10.2: No caption for this figure - Figure 10.7
- 10.2: Change “draws” to “drawn” - As expected, the fit lines draws by the model with cubic polynomials and an interaction do well in capturing the distinct patterns, in the two M subsets.
- 10.2: The slope in this subset of the domain for the quadratic fit is negative so should lead to a decrease in Y - When M=0 and X~3, increasing X by a small amount will result in a substantial (and statistically significant) increase in Y.

**Comment by vincentarelbundock** [comment]
wow, you're really going through this whole thing! Thanks, this is super useful. And just in time for me to correct the proofs, which should be coming soon.

**Comment by vincentarelbundock** [comment]
Thanks!

**Comment by timchisamore** [comment]
- 16.2.2: Should “)” be “avg_slopes()”? - Use [slopes()](https://marginaleffects.com/man/r/slopes.html) to [compute marginal effects (slopes of the regression equation)](https://marginaleffects.com/chapters/slopes.html) for each row of the dataset, and use ) to compute “Average Marginal Effects”, that is, the average of all observation-level marginal effects:
- 16.3.1: “residual” repeated - We also use the type argument to compute two types of predictions: accounting for residual (observation-level) residual variance (prediction) or ignoring it (response).
- 16.3.6: “NA” should be “New Region” - We can also obtain predictions or marginal effects for a hypothetical group instead of one of the observed regions. To achieve this, we create a dataset with NA in the region column. 
- 16.4.1: I believe this example is using the Heating dataset and not mtcars - Use the plot_predictions function to plot conditional adjusted predictions for each level of the outcome variable gear, conditional on the value of the mpg regressor:
- 18.3: This sentence was never completed - We thus Thus,
- 18.3: Add “in” between “interested” and “comparison” - Since we are not interested comparison pairs where both profiles have the same language skills, we use the [subset()](https://rdrr.io/r/base/subset.html) function with the newdata argument to select an appropriate grid.
- 19: Consider defining “eyex”, “eydx”, and “dyex” - We can estimate the “eyex”, “eydx”, and “dyex” (semi-)elasticities with respect to as follows:
- 20.1.1: Should “operates” be “operators” or “operations” - Here, we create a function called lm_manual(), which estimates a linear regression model using simple linear algebra operates:
- 20.1.2: “Model" should be “models” - Note that, for custom model, we typically have to supply values for the newdata and variables arguments explicitly.
- 21.3: “Slop” should be “slope” - The [slopes()](https://marginaleffects.com/man/r/slopes.html) function produces a dataset with the same number of rows as the original data, but with new columns for the slop and uncertainty estimates:
- 22: Should this be “cATE” or “CATE” - This exercise inspired by [Scholbeck et al. 2022](https://arxiv.org/abs/2201.08837) and their concept of “cATE”.

**Comment by timchisamore** [comment]
- 24: Add “on” between “relies” and “only” - ITS is arguably even more difficult to get right, since it typically relies only a single times series, and thus lacks a true control group.
- 27.1.1: More than 6 are displayed - Here, we display only the first 6 of them:
- 27.1.2: Add “as” between “differences” and “an” and remove “a” - We can present the entire distribution of unit-level proportion differences an a cumulative distribution function:
- 29.3: It might be worthwhile explaining why this variance—covariance structure is required - [avg_comparisons](https://marginaleffects.com/man/r/comparisons.html)(fit, variables = "treat", newdata = [subset](https://rdrr.io/r/base/subset.html)(dat, treat == 1), vcov = ~subclass, wts = "weights")
- 31.5: Were the uses of “/“ before and after “different” intentional - However, if we wish to supply each model in our list resulting after multiple imputation with a /different/ dataset on which to calculate results, we cannot use newdata.
- 33.2: It looks like you intended to add what the most costly operation was but did not - As noted above, the most costly operation in marginaleffects, because that involves calling [predict()](https://rdrr.io/r/stats/predict.html) at least twice for every coefficient in the model.
- 38.1: This is the first sentence of the case study, so there is nothing above - For linear mixed effects models we can apply the Satterthwaite and Kenward-Roger corrections in the same way as above:
- 38.2: Considering using either e.g. or etc. but not both in the same sentence - where we take the derivative with respect to a model’s coefficients, and g is a marginaleffects function which returns some quantity of interest (e.g., slope, contrasts, predictions, etc.)

**Comment by vincentarelbundock** [comment]
🔥🔥🔥🔥🔥🔥

**Comment by timchisamore** [comment]
- 34: The section starts as 34 but ends up as 36, should these all be 34 with the use subsections such as 34.1.1, 34.2, etc.
- 34: Add “an” between “as” and “independent” - For the analysis of the data we will first fit a standard Cox proportional hazards model with the time to death or right-censoring as the response variable and the hormonal treatment as independent variable. 
- 34: Consider changing “including” to “indicating” - Additionally, we will include grade, which is an ordinal variable reflecting the level of tumor differentiation, and age at surgery, a continuous variable including the age in years as further independent variables. 
- 34: it should be hormon=0 and hormon=1- At time t, the difference between the estimated survival probability for patients in group hormon=1 and hormon=1 is equal to zero.
- 34: Consider rearranging to “One important challenge arises because the log-likelihood of the Cox proportional-hazards model …”- One important challenge arises because the Cox proportional‐hazards model the log‑likelihood naturally factorizes into a partial component that involves only the regression coefficients and a baseline component that involves the cumulative hazard.
- 34: Remove “can that we”- The key innovation that underpins this model is that we can that we do not need to optimize the likelihood of the full model.
- 34: Change “hazard” to “hazards” - Instead, if we are willing to accept some assumptions (i.e., proportional hazard), we can proceed with the partial likelihood.
- 34: The section starts as 34 but ends up as 36, should these all be 34 with the use subsections such as 34.1.1, 34.2, etc.
- 34: Add “an” between “as” and “independent” - For the analysis of the data we will first fit a standard Cox proportional hazards model with the time to death or right-censoring as the response variable and the hormonal treatment as independent variable. 
- 34: Consider changing “including” to “indicating” - Additionally, we will include grade, which is an ordinal variable reflecting the level of tumor differentiation, and age at surgery, a continuous variable including the age in years as further independent variables. 
- 34: it should be hormon=0 and hormon=1- At time t, the difference between the estimated survival probability for patients in group hormon=1 and hormon=1 is equal to zero.
- 34: Consider rearranging to “One important challenge arises because the log-likelihood of the Cox proportional-hazards model …”- One important challenge arises because the Cox proportional‐hazards model the log‑likelihood naturally factorizes into a partial component that involves only the regression coefficients and a baseline component that involves the cumulative hazard.
- 34: Remove “can that we”- The key innovation that underpins this model is that we can that we do not need to optimize the likelihood of the full model.
- 34: Change “hazard” to “hazards” - Instead, if we are willing to accept some assumptions (i.e., proportional hazard), we can proceed with the partial likelihood.

**Comment by vincentarelbundock** [comment]
great stuff, as always!

**Comment by vincentarelbundock** [comment]
sent to the publisher.

Thanks again!
